src/components/diagrams/diagram-pane.js

```diff
@@ -286,95 +362,314 @@ export function DiagramPane({
-                        ...emptyDiagramView,
-                        align: 'left',
+                // Then we add the data when the fetch is finished
+                diagramsToAdd.forEach((diagramState) => {
+                    createView(diagramState).then((singleLineDiagramView) => {
+                        setViews((views) => {
+                            const diagramViewId = views.findIndex(
+                                (view) =>
+                                    view.svgType !==
+                                        DiagramType.NETWORK_AREA_DIAGRAM &&
+                                    view.id === diagramState.id
+                            );
+                            const updatedViews = views.slice();
+                            // we update the SLD with the fetched data
+                            updatedViews[diagramViewId] = {
+                                ...updatedViews[diagramViewId],
+                                ...singleLineDiagramView,
+                                loadingState: false,
+                            };
+                            return updatedViews;
+                        });
                     });
+                });
+            }
+        },
+        [createView, intl]
+    );
+
+    // Check if we need to remove old SLDs from the 'views' and remove them if necessary
+    const removeSLDs = useCallback((diagramStates) => {
+        // We check if we need to remove old diagrams
+        const diagramIdsToRemove = [];
+        viewsRef.current.forEach((diagramView) => {
+            if (diagramView.svgType !== DiagramType.NETWORK_AREA_DIAGRAM) {
+                const diagramStillPresentInRedux = diagramStates.find(
+                    (diagramState) =>
+                        diagramState.svgType !==
+                            DiagramType.NETWORK_AREA_DIAGRAM &&
+                        diagramState.id === diagramView.id
+                );
+                if (!diagramStillPresentInRedux) {
+                    diagramIdsToRemove.push(diagramView.id);
                 }
             }
         });
 
-        if (networkAreaIds.length > 0) {
-            let networkAreaDiagramView = createView({
+        // If we found diagrams to remove, we remove them
+        if (diagramIdsToRemove?.length) {
+            setViews((views) => {
+                const updatedViews = views.filter(
+                    (view) =>
+                        view.svgType === DiagramType.NETWORK_AREA_DIAGRAM ||
+                        !diagramIdsToRemove.includes(view.id)
+                );
+                return updatedViews;
+            });
+        }
+    }, []);
+
+    // Check if we need to remove or add SLDs
+    const updateSLDs = useCallback(
+        (diagramStates) => {
+            removeSLDs(diagramStates);
+            addSLDs(diagramStates);
+        },
+        [removeSLDs, addSLDs]
+    );
+
+    const addOrReplaceNAD = useCallback(
```

Comment:
A simple comment is welcome here to help understand the goal of this function

Reply:
Ok done